src/main/kotlin/com/openlattice/data/DataGraphService.kt

```diff
@@ -289,78 +283,45 @@ open class DataGraphService(
         return entityWriteEvent
     }
 
-    private fun deleteAssociations(entitySetId: UUID, entityKeyIds: Optional<Set<UUID>>) {
-        // collect association entity key ids
-        val associationsEntityKeyIds = collectAssociationsEntityKeyIds(entitySetId, entityKeyIds)
-
-        // access checks
-        val authorizedPropertyTypesOfAssociations = HashMap<UUID, Map<UUID, PropertyType>>()
-        associationsEntityKeyIds.forEach {
-            val authorizedPropertyTypesOfAssociation = getAuthorizedPropertyTypesForDelete(it.key, Optional.empty())
-            authorizedPropertyTypesOfAssociations[it.key] = authorizedPropertyTypesOfAssociation
-        }
-
-        // delete associations of entity set
-        val associationDeleteCount = associationsEntityKeyIds.map {
-            deleteEntityDataAnVertices(
-                    it.key,
-                    it.value,
-                    authorizedPropertyTypesOfAssociations[it.key]!!).numUpdates
-        }.sum()
-
-        logger.info("Deleted {} associations when deleting entities from entity set {}",
-                associationDeleteCount, entitySetId)
-    }
-
-    private fun collectAssociationsEntityKeyIds(
-            entitySetId: UUID, entityKeyIds: Optional<Set<UUID>>): Map<UUID, Set<UUID>> {
-        val associations = if (entityKeyIds.isPresent) {
-            getEdgesConnectedToEntities(entitySetId, entityKeyIds.get())
-        } else {
-            getEdgeKeysOfEntitySet(entitySetId)
-        }
-        val associationsEntityKeyIds = HashMap<UUID, MutableSet<UUID>>()
-        associations.forEach {
-            val previousValue = associationsEntityKeyIds
-                    .putIfAbsent(it.edge.entitySetId, mutableSetOf(it.edge.entityKeyId))
-            if (previousValue != null) {
-                associationsEntityKeyIds[it.edge.entitySetId]!!.add(it.edge.entityKeyId)
+    override fun deleteAssociationsBatch(
+            entitySetId: UUID,
+            associationsEdgeKeys: PostgresIterable<EdgeKey>,
+            authorizedPropertyTypes: Map<UUID, Map<UUID, PropertyType>>): Set<WriteEvent> {
+        var associationDeleteCount = 0
+        val edgeKeysIterator = associationsEdgeKeys.iterator()
+        val writeEvents = HashSet<WriteEvent>()
+
+        while (edgeKeysIterator.hasNext()) {
```

Comment:
I think you borrowed this pattern from linker. For various reasons I think you might benefit from doing ```associationEdgeKeys.asSequence().chunked(...)``` here to read a fixed number of items from the underlying iterable.

It's important to make sure the calling iterable has fetchSize set so it doesn't load the full thing into memory (i.e Postgres will stream data).